pkg/sdn/plugin/node.go

```diff
@@ -187,29 +190,123 @@ func (node *OsdnNode) dockerPreCNICleanup() error {
 	}
 
 	// Wait until docker has restarted since kubelet will exit it docker isn't running
-	dockerClient, err := docker.NewClientFromEnv()
-	if err != nil {
-		return fmt.Errorf("failed to get docker client: %v", err)
+	if _, err := ensureDockerClient(); err != nil {
+		return err
 	}
-	err = kwait.ExponentialBackoff(
+
+	log.Infof("Cleaned up left-over openshift-sdn docker bridge and interfaces")
+
+	return nil
+}
+
+func ensureDockerClient() (dockertools.DockerInterface, error) {
+	endpoint := os.Getenv("DOCKER_HOST")
+	if endpoint == "" {
+		endpoint = "unix:///var/run/docker.sock"
+	}
+	dockerClient := dockertools.ConnectToDockerOrDie(endpoint, time.Minute, time.Minute)
+
+	// Wait until docker has restarted since kubelet will exit it docker isn't running
+	if err := kwait.ExponentialBackoff(
 		kwait.Backoff{
 			Duration: 100 * time.Millisecond,
 			Factor:   1.2,
 			Steps:    6,
 		},
 		func() (bool, error) {
-			if err := dockerClient.Ping(); err != nil {
+			if _, err := dockerClient.Version(); err != nil {
 				// wait longer
 				return false, nil
 			}
 			return true, nil
-		})
+		}); err != nil {
+		return nil, fmt.Errorf("failed to connect to docker after SDN cleanup restart: %v", err)
+	}
+	return dockerClient, nil
+}
+
+func formatPod(pod *kapi.Pod) string {
+	return fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+}
+
+func dockerSandboxNameToInfraPodNamePrefix(name string) (string, error) {
```

Comment:
this is clearly copied from somewhere in the guts of k8s and should probably indicate where, so if it breaks in the future we know where to look